.github/workflows/release-integration.yml

```diff
@@ -0,0 +1,94 @@
+  RELEASE_NOTES_FILE: "RELEASE-BODY.md"
+jobs:
+  pre-release:
+    runs-on: ubuntu-20.04
+    outputs:
+      tag-name: ${{ steps.create-release-package.outputs.tag-name }}
+    steps:
+      - name: Checkout repo
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+
+      - name: Set up Node.js
+        uses: actions/setup-node@v2
+        with:
+          node-version: ${{ env.NODE_VERSION }}
+
+      - name: Configure Git
+        run: |
+          git config user.name "GitHub Actions" # "${{ github.actor }}"
+          git config user.email "actions@github.com" # "${{ github.actor }}@users.noreply.github.com"
+
+      - name: Prepare GitHub Release Notes
+        run: |
+          # Delete pre-release tags to be able to generate a changelog from last 'real' release LOCALLY
+          # This is a workaround for a known limitation of standard-version
+          # Reference: https://github.com/conventional-changelog/standard-version/issues/203#issuecomment-872415140
+          git tag -l | grep -vE '^(0|[1-9][0-9]*)\.(0|[1-9][0-9]*)\.(0|[1-9][0-9]*)$' | xargs git tag -d
+
+          # generate release notes into RELEASE_NOTES_FILE
+          npx standard-version@^9.3.1 -i "${{ env.RELEASE_NOTES_FILE }}" --skip.commit --skip.tag --header ""          
+
+      - name: Create release package
+        id: create-release-package
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: |
+          echo "🚀 Creating release package now..."
+          # this will generate a commit and a tag
+          npx standard-version@^9.3.1 --skip.commit
+
+          TAG=$(git describe --tags --abbrev=0)
+
+          # determine tag name
+          echo "::set-output name=tag-name::$TAG"
+
+          # create a commit with the changelog
+          git add CHANGELOG.md
+          git commit -s -m "chore(release): $TAG"
+
+      - name: Push changes to repo
+        id: push-changes
+        env:
+          RELEASE_TAG: ${{ steps.create-release-package.outputs.tag-name }}
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: |
+          echo "Fetching previously deleted old tags..."
+          git fetch origin --tags -f
+          TARGET_BRANCH=patch/release-notes-$RELEASE_TAG
+          # delete existing branch just in case
+          git branch -D $TARGET_BRANCH &>/dev/null || true
+          # create new branch
+          git checkout -b $TARGET_BRANCH
+          # push changes
+          echo "⚡️ Pushing changes to remote repository..."
+          git push -f --follow-tags --set-upstream origin patch/release-notes-$RELEASE_TAG
+
+
+      - name: Create GitHub Pull Request for release notes
+        env:
+          RELEASE_TAG: ${{ steps.create-release-package.outputs.tag-name }}
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: |
+          gh pr create --title "chore: Release notes for $RELEASE_TAG" --body "**This is an automated PR for release notes of $RELEASE_TAG!**"
```

Comment:
Why not use the `peter-evans/create-pull-request@v3` action for doing all this manual stuff? I think it would make this code more readable.

Reply:
Several reasons:
* I ran into a problem with `peter-evans/create-pull-request@v3` which I couldn't fix easily
* Using the `gh` CLI we are independent from another third party vendor (yes, `gh` classifies as a third party vendor too, but we are already using it to create the release)
* IMO the code is better readable if we refer to the `gh` CLI as often as we can (it creates a repetitive pattern), see screenshot:
![image](https://user-images.githubusercontent.com/56065213/144380299-ccd24510-6b51-40c4-a140-7fdf0ed828b2.png)

Reply:
In terms of readability, the GH CLI is not my issue but rather all the bash code before that :)
I feel like with a GH action, the variable names immediately show exactly what branch is used, what repo is used, what commit message is used, etc.

Reply:
The bash code used in the lines before that line is necessary because `standard-version` does not have an option to sign a commit using `git commit -s`. They only have the option to sign it using `git commit -S`, which uses GPG, which is not available unless we configure a certificate for the gh action user (which is absolutely out of scope for this feature).

Signing the commit is necessary because DCO bot requires it and does not make an exception for the gh-action user (also, this is out of scope for this feature).

Long story short, I would suggest using the code as it is, creating an Issue in `standard-version` regarding the signing issue, and refer to the implementation of this issue in our code.

Reply:
https://github.com/conventional-changelog/standard-version/issues/854

Reply:
Ok I guess then it's fine